pkg/controller/nodeipam/ipam/multi_cidr_range_allocator.go

```diff
@@ -1090,19 +1163,22 @@ func (r *multiCIDRRangeAllocator) mapClusterCIDRSet(cidrMap map[string][]*cidrse
 	return nil
 }
 
-// reconcileDelete deletes the ClusterCIDR object and removes the finalizer.
+// reconcileDelete releases the assigned ClusterCIDR and removes the finalizer
+// if the deletion timestamp is set.
 func (r *multiCIDRRangeAllocator) reconcileDelete(clusterCIDR *networkingv1alpha1.ClusterCIDR) error {
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
-	if slice.ContainsString(clusterCIDR.GetFinalizers(), clusterCIDRFinalizer, nil) {
+	if !clusterCIDR.GetDeletionTimestamp().IsZero() && slice.ContainsString(clusterCIDR.GetFinalizers(), clusterCIDRFinalizer, nil) {
+		klog.V(2).Infof("Releasing ClusterCIDR: %s", clusterCIDR.Name)
 		if err := r.deleteClusterCIDR(clusterCIDR); err != nil {
+			klog.V(2).Infof("Error while deleting ClusterCIDR: %+v", err)
 			return err
 		}
 		// Remove the finalizer as delete is successful.
 		cccCopy := clusterCIDR.DeepCopy()
 		cccCopy.ObjectMeta.Finalizers = slice.RemoveString(cccCopy.ObjectMeta.Finalizers, clusterCIDRFinalizer, nil)
-		if _, err := r.client.NetworkingV1alpha1().ClusterCIDRs().Update(context.TODO(), clusterCIDR, metav1.UpdateOptions{}); err != nil {
+		if _, err := r.client.NetworkingV1alpha1().ClusterCIDRs().Update(context.TODO(), cccCopy, metav1.UpdateOptions{}); err != nil {
```

Comment:
upps

Reply:
Is there any concern with this change or is it for how we missed it before 😛